packages/logs/src/domain/logger.ts

```diff
@@ -39,29 +51,47 @@ export class Logger {
   }
 
   @monitored
-  log(message: string, messageContext?: object, status: StatusType = StatusType.info) {
-    this.handleLogStrategy({ message, context: deepClone(messageContext) as Context, status }, this)
+  log(message: string, messageContext?: object, status: StatusType = StatusType.info, error?: Error) {
+    let context: Context
+
+    if (status === StatusType.error || (error !== undefined && error !== null)) {
+      // Always add origin if status is error (backward compatibility), or we have an actual error
+      const errorContext = { error: { origin: ErrorSource.LOGGER } as Context }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏typings improvements + postponing the `error` field use
```suggestion
      const errorContext = ({ origin: ErrorSource.LOGGER } as LogsEvent['error'])!
      ...
      errorContext.kind = stackTrace.name
      ...
      context = combine({ error: errorContext }, messageContext)
```

Reply:
👌